resources/guidelines/code/backward-compatibility.md

```diff
@@ -0,0 +1,751 @@
+
+## Annotations
+During the development different cases occur, where you want to replace old code with new one or even remove some obsolete code. As Shopware must always be backward compatible on minor and patch updates, old code should only be removed with a major release. Until the next major release you want to mark the code with a corresponding annotation, to inform other developers of the planned change. This overview shows the most important annotations and in which situation they have to be used.
+
+### @deprecated
+```php
+/**
+ * @deprecated tag:v6.5.0 - Use NewFunction() instead
+ */
+```
+The `@deprecated` annotation is used for obsolete public code, which will be removed with the next major release. The annotation always needs the specific major version tag, in which the code should be removed. Always add a meaningful comment with information about the corresponding replacement and how the deprecated code can be removed.
+
+### @feature-deprecated
+```php
+/**
+ * @feature-deprecated (flag:FEATURE_NEXT_11111)
+ */
+```
+The `@feature-deprecated` annotation is used for obsolete code during the development of a feature, when the code is still hidden behind a feature flag. This is important, so that code, that is not public, will not trigger deprecation warnings. This annotation has to be changed to the correct `@deprecated` annotation, when the feature is released and the corresponding feature flag is removed. Always add the name of the corresponding feature flag to the annotation, so that it will not be forgotten, when the flag is removed.
+
+### @major-deprecated
+```php
+/**
+ * @major-deprecated (flag:FEATURE_NEXT_22222)
+ */
+```
+The `@major-deprecated` annotation is used for breaking code which has to stay behind a specific major feature flag until the next major release. Always add the name of the corresponding feature flag to the annotation, so that it will not be forgotten, when the flag is removed.
+
+### @internal
+```php
+/**
+ * @internal (flag:FEATURE_NEXT_11111)
+ */
+```
+The `@internal` annotation is used for new introduced code, which is not yet released or hidden behind a feature flag. This ensures, that it will not be treated as public API until the corresponding feature is released and makes it possible to change the code in any way until the final release. Always add the name of the corresponding feature flag to the annotation, so that it will not be forgotten, when the corresponding feature is released.
+
+## Workflows
+### Backward Compatible Features
+When developing new features, the goal should always be, to do this in a backward compatible way. This ensures that the feature can be shipped with a minor release, to provide value for customers as soon as possible. The following table should help you to use the correct approach for each type of change.
+
+<table>
```

Comment:
Markdown-Table. [Gitbook Markdown](https://docs.gitbook.com/editing-content/markdown).

Reply:
I've converted all tables to markdown, it just looks broken in the github diff because of the enforced soft-wrap of long lines.